plinth/modules/dynamicdns/dynamicdns.py

```diff
@@ -0,0 +1,352 @@
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from django import forms
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.core import validators
+from django.core.urlresolvers import reverse_lazy
+from django.template.response import TemplateResponse
+from gettext import gettext as _
+import logging
+
+from plinth import actions
+from plinth import cfg
+from plinth import package
+
+LOGGER = logging.getLogger(__name__)
+
+subsubmenu = [{'url': reverse_lazy('dynamicdns:index'),
+               'text': _('About')},
+              {'url': reverse_lazy('dynamicdns:configure'),
+               'text': _('Configure')},
+              {'url': reverse_lazy('dynamicdns:statuspage'),
+               'text': _('Status')}
+              ]
+
+
+def init():
+    """Initialize the dynamicdns module"""
+    menu = cfg.main_menu.get('apps:index')
+    menu.add_urlname('Dynamic DNS', 'glyphicon-refresh',
+                     'dynamicdns:index', 40)
+
+
+@login_required
+@package.required('ez-ipupdate')
+def index(request):
+    """Serve dynamic DNS  page"""
+
+    index_subsubmenu = subsubmenu
+
+    return TemplateResponse(request, 'dynamicdns.html',
+                            {'title': _('dynamicdns'),
+                             'subsubmenu': index_subsubmenu})
+
+
+class TrimmedCharField(forms.CharField):
+    """Trim the contents of a CharField"""
+    def clean(self, value):
+        """Clean and validate the field value"""
+        if value:
+            value = value.strip()
+
+        return super(TrimmedCharField, self).clean(value)
+
+
+class ConfigureForm(forms.Form):
+    """Form to configure the dynamic DNS client"""
+
+    hlp_updt_url = 'The Variables &lt;User&gt;, &lt;Pass&gt;, &lt;Ip&gt;, \
+                    &lt;Domain&gt; may be used.'
+
+    hlp_services = 'Please choose a update protocol according to your \
+                    provider. If your provider does not support the GnudIP \
+                    protocol or your provider is not listed you may use \
+                    the update URL of your provider.'
+
+    enabled = forms.BooleanField(label=_('Enable Dynamic DNS'),
+                                 required=False, widget=forms.CheckboxInput
+                                (attrs={'onclick': 'mod_form();'}))
+
+    service_type = forms.ChoiceField(label=_('Service type'),
+                                     help_text=_(hlp_services),
+                                     choices=(('1', 'GnuDIP'),
+                                             ('2', 'noip.com'),
+                                             ('3', 'selfhost.bz'),
+                                             ('4', 'other Update URL')),
+                                     widget=forms.Select(attrs={'onChange':
```

Comment:
Don't put javascript into python files, but in the template or into a separate .js file if it gets a lot of code.